### PR TITLE
feat(react): add useUser react hook

### PR DESCRIPTION
**Pull request by andreribeirosilva**
## Description

This PR is part of the work being developed in order to achieve consistent interfaces among all the react hooks (provided in the `blackout-react` package).

The `useUser` hook is now available exposing the following interface:

- Return
    - isUserLoading
    - isLoginLoading
    - isLogoutLoading
    - isRegisterLoading
    - isChangePasswordLoading
    - isRecoverPasswordLoading
    - isResetPasswordLoading
    - userError
    - loginError
    - logoutError
    - registerError
    - changePasswordError
    - recoverPasswordError
    - resetPasswordError
    - authenticationError
    - isFetched
    - data
        - … user (user entities data)
    - actions
        - login
        - logout
        - register
        - changePassword
        - resetPassword
        - recoverPassword
        - fetch
        - update

On this MR, the wishlist's types files were also renamed in order to add the prefix `types`

### Dependencies

None

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/react/src/users/hooks/useUser.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  getAuthenticationError,
+  getChangePasswordError,
+  getLoginError,
+  getLogoutError,
+  getRecoverPasswordError,
+  getRegisterError,
+  getResetPasswordError,
+  getUser,
+  getUserError,
+  isAuthenticationLoading as isAuthenticationLoadingSelector,
```
Comment (boliveira) [inline packages/react/src/users/hooks/useUser.ts:13]: `isAuthenticationLoading` selector is now just an alias for `isUserLoading` as they were reflecting the same thing previously. I think we can live with just `isUserLoading` and discard using the `isAuthenticationLoading` selector. What do you think?

File: packages/react/src/users/hooks/types/useUser.types.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import type { Config } from '@farfetch/blackout-client';
+
+export type UseUSerOptions = {
```
Comment (boliveira) [inline packages/react/src/users/hooks/types/useUser.types.ts:3]:
```suggestion
export type UseUserOptions = {
```

File: packages/react/src/users/hooks/useUser.ts (hunk 1 above)
Comment (andreribeirosilva) [reply packages/react/src/users/hooks/useUser.ts:13]: Yep! will remove it 👌

**Review by PedroMiguelSS — APPROVED** [review]
LGTM 👍

**Review by boliveira — APPROVED** [review]

**Pull request merged**